### DISCUSSION TITLE
Add embed functionality

### DISCUSSION THREAD

**Discussion by ghost** [odin-lang/Odin#1367] [Ideas/Requests]
It would be very useful if there were functionality that lets you embed files into your program, similar to Go's use of it with //go:embed, or Nim's `os.slurp()`(staticRead), and then allows you to access the file as if it were in the same directory as the Odin program.

It could possibly be something like @(embed="file.txt") above a const variable, embedding the given file, `file.txt` into the variable.

**Comment by Kelimion** [comment]
`#load(filename)` will embed a file and make it available as a `[]u8` constant.

**Reply by ghost** [reply]
Oh, nice. Do I have to import anything to use it? Where's the documentation for it? I couldn't find anything relating to embedding files.

**Reply by Kelimion** [reply]
Nothing to import. It's a directive. Good idea to add it to the [docs](https://odin-lang.org/docs/overview/).

There's 3 related ones:
`#load(filename)` returns a constant slice of bytes, the file contents.
`#load_or(filename) or alternative_constant` returns a constant slice of bytes, the file contents, or another `[]byte` constant if the file wasn't present.
`#load_hash(filename, algorithm)` returns a compile-time constant hash of the file.
